mistbigqueryuploader.js

```diff
@@ -1,106 +1,139 @@
         option: "--gauth",
-        type: "str"
-      }
-    }
+        type: "str",
+      },
+    },
   };
   process.stdout.write(JSON.stringify(capa) + "\n");
   return 0;
 }
-if (argv.port){process.env.PORT = argv.port;}
-if (argv.addr){process.env.ADDR = argv.addr;}
-if (argv.dataset){process.env.DATASET = argv.dataset;}
-if (argv.table){process.env.TABLE = argv.table;}
-if (argv.gauth){process.env.GOOGLE_APPLICATION_CREDENTIALS = argv.gauth;}
+if (argv.port) {
+  process.env.PORT = argv.port;
+}
+if (argv.addr) {
+  process.env.ADDR = argv.addr;
+}
+if (argv.dataset) {
+  process.env.DATASET = argv.dataset;
+}
+if (argv.table) {
+  process.env.TABLE = argv.table;
+}
+if (argv.gauth) {
+  process.env.GOOGLE_APPLICATION_CREDENTIALS = argv.gauth;
+}
 
 // Check environment
-if (!process.env.PORT){throw new Error("PORT environment variable not set!");}
-if (!process.env.ADDR){throw new Error("ADDR environment variable not set!");}
-if (!process.env.DATASET){throw new Error("DATASET environment variable not set!");}
-if (!process.env.TABLE){throw new Error("TABLE environment variable not set!");}
+if (!process.env.PORT) {
+  throw new Error("PORT environment variable not set!");
+}
+if (!process.env.ADDR) {
+  throw new Error("ADDR environment variable not set!");
+}
+if (!process.env.DATASET) {
+  throw new Error("DATASET environment variable not set!");
+}
+if (!process.env.TABLE) {
+  throw new Error("TABLE environment variable not set!");
+}
 
 // Create listening UDP socket
-const s = udp.createSocket('udp4');
-s.on('listening', () => {
+const s = udp.createSocket("udp4");
+
+s.on("listening", () => {
   const address = s.address();
   console.log(`Waiting for UDP packets on ${address.address}:${address.port}`);
 });
-s.on('message', (msg, rinfo) => {
+
+s.on("message", (msg, rinfo) => {
   let json = false;
-  try{
+  const date = new Date();
+  try {
     json = JSON.parse(msg);
-  }catch(e){
+  } catch (e) {
     console.log(`Ignored invalid JSON: ${msg}`);
   }
   // Inject extra user agent info
-  if ("user_agent" in json){
+  if ("user_agent" in json) {
     let ua = uap(json["user_agent"]);
     json["ua"] = {
-      "result": ua,
-      "os": ua.os.name,
-      "engine": ua.engine.name,
-      "device": ua.device
+      result: ua,
+      os: ua.os.name,
+      engine: ua.engine.name,
+      device: ua.device,
     };
   }
-  bq.dataset(process.env.DATASET).table(process.env.TABLE).insert([{log: JSON.stringify(json), timestamp: Date.now(), id: crypto.randomUUID()}]).catch((e) => {
-    console.warn(`${e}`);
-    if (e.errors){
-      for (err in e.errors){
-        console.warn(e.errors[err]);
-      }
-    }
-  });
+  for (const tableName of process.env.TABLE.split(",")) {
+    bq.dataset(process.env.DATASET)
+      .table(tableName)
+      .insert([
+        {
+          log: JSON.stringify(json),
+          id: crypto.randomUUID(),
+          timestamp: date.getTime(),
+          timestamp_ts: bq.timestamp(date),
```

Comment:
we don't want this column on the original table. 

the idea is to write to the original one with no changes and the new one with the new column. we'll do this for a couple of days while we change over the data pipeline and then we can remove the original table.